src/pybind/mgr/dashboard/frontend/src/app/ceph/nfs/nfs-list/nfs-list.component.ts

```diff
@@ -150,6 +151,11 @@ export class NfsListComponent implements OnInit, OnDestroy {
         );
       },
       (error) => {
+        if (error.error.detail.includes('GANESHA_RADOS_POOL_NAMESPACE')) {
+          error.preventDefault();
+          this.notConfigured = true;
+        }
```

Comment:
Could you please add some test?

As error testing can be somewhat tricky I wrote 2 example tests in advanced.

```diff
diff --git a/src/pybind/mgr/dashboard/frontend/src/app/ceph/nfs/nfs-list/nfs-list.component.spec.ts b/src/pybind/mgr/dashboard/frontend/src/app/ceph/nfs/nfs-list/nfs-list.component.spec.ts
index 056e6b45b0..3d9c5687b0 100644
--- a/src/pybind/mgr/dashboard/frontend/src/app/ceph/nfs/nfs-list/nfs-list.component.spec.ts
+++ b/src/pybind/mgr/dashboard/frontend/src/app/ceph/nfs/nfs-list/nfs-list.component.spec.ts
@@ -5,7 +5,7 @@ import { RouterTestingModule } from '@angular/router/testing';
 
 import { ToastModule } from 'ng2-toastr';
 import { TabsModule } from 'ngx-bootstrap/tabs';
-import { BehaviorSubject, of } from 'rxjs';
+import { BehaviorSubject, of, throwError } from 'rxjs';
 
 import {
   configureTestBed,
@@ -63,6 +63,37 @@ describe('NfsListComponent', () => {
     expect(component).toBeTruthy();
   });
 
+  describe('fetch errors', () => {
+    let resp;
+    let stoppedNotification;
+
+    const testError = (errorDetail, notConfigured) => {
+      resp.error.detail = errorDetail;
+      fixture.detectChanges();
+      expect(component.notConfigured).toBe(notConfigured);
+      expect(stoppedNotification).toBe(notConfigured);
+      expect(component.onFetchError).toHaveBeenCalled();
+    };
+
+    beforeEach(() => {
+      spyOn(component, 'onFetchError');
+      stoppedNotification = false;
+      resp = {
+        error: { detail: '' },
+        preventDefault: () => (stoppedNotification = true)
+      };
+      spyOn(nfsService, 'daemon').and.callFake(() => throwError(resp));
+    });
+
+    it('test normal error', () => {
+      testError('Something gone wrong', false);
+    });
+
+    it('test component not configured error', () => {
+      testError('Oh no GANESHA_RADOS_POOL_NAMESPACE is not set', true);
+    });
+  });
+
   describe('after ngOnInit', () => {
     beforeEach(() => {
       fixture.detectChanges();
```

Reply:
Change was reverted.